### PR TITLE
model/endpoints for the schedule entity 

### LINKED ISSUE #39: Model entity shedule and create enpoints for that same entity
Create the model for the entity schedule and the respective repository, services and endpoints on spring boot

### PR DESCRIPTION
**Pull request by DiogoSilva1904**
### Issue

Closes #39 

### Reason for this change

Modeling the entity schedule, and making the respective service,repository and controller.

### Description of changes

The entity schedule has been created along with its corresponding repository, services, and controllers. The controllers handle the following operations such as retrieving all schedules, fetching schedules by their unique IDs, deleting schedules, saving new schedules, and updating existing ones.Swagger documentation already done for the this controllers. 

### Description of how you validated changes

There were made unitary tests for the repository, services and endpoints

### Checklist
- [x] I have tested my changes locally.
- [x] I have updated the documentation to reflect my changes.
- [x] I have added/updated unit tests (if applicable).
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://ua-smart-signage-platform.github.io/Documentation-Website/guidelines/)

### CODE REVIEW

**Review by Rafa548 — CHANGES_REQUESTED** [review]
Change PR description, still a minor issue in the sonarcloud

**Review by Rafa548 — APPROVED** [review]

**Comment by Rafa548** [comment]
lgtm

File: backend/uasmartsignage/src/main/java/deti/uas/uasmartsignage/Models/Schedule.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+@Entity
+@Table(name = "schedule")
+public class Schedule {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(name = "frequency")
+    private String frequency;
```
Comment (fungame2270) [inline backend/uasmartsignage/src/main/java/deti/uas/uasmartsignage/Models/Schedule.java:25]: change to int

File: backend/uasmartsignage/src/main/java/deti/uas/uasmartsignage/Models/Schedule.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Column(name = "n_times")
+    private int nTimes;
```
Comment (fungame2270) [inline backend/uasmartsignage/src/main/java/deti/uas/uasmartsignage/Models/Schedule.java:28]: this is not needed in bd

File: backend/uasmartsignage/src/main/java/deti/uas/uasmartsignage/Models/Schedule.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    @JoinColumn(name = "created_by")
+    private User createdBy;
+
+    @ManyToOne
+    @JoinColumn(name = "last_edited_by")
+    private User lastEditedBy;
+
+    @Column(name = "created_on")
+    private LocalDate createdOn;
+
```
Comment (fungame2270) [inline backend/uasmartsignage/src/main/java/deti/uas/uasmartsignage/Models/Schedule.java:55]: need to add start hour and end hour
and a list for weekday its a list of ints postgres accepts lists in bd

(the created on is good i dont know why its selected)

File: backend/uasmartsignage/src/main/java/deti/uas/uasmartsignage/Models/Schedule.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    private int intervalOfTime;
+
+    @Column(name = "start_date")
+    private LocalDateTime startDate;
+
+    @Column(name = "end_date")
+    private LocalDateTime endDate;
+
+    @Column(name = "date")
+    private LocalDate date;
```
Comment (fungame2270) [inline backend/uasmartsignage/src/main/java/deti/uas/uasmartsignage/Models/Schedule.java:40]: this date is not needed

**Review by fungame2270 — COMMENTED** [review]
please make this changes

**Comment by DiogoSilva1904** [comment]
This changes will be adressed in a new issue. #102

File: backend/uasmartsignage/src/main/java/deti/uas/uasmartsignage/Models/Schedule.java (hunk 3 above)
Comment (DiogoSilva1904) [reply backend/uasmartsignage/src/main/java/deti/uas/uasmartsignage/Models/Schedule.java:55]:
```java
@Column(name = "start_date")
   private LocalDateTime startDate;

   @Column(name = "end_date")
    private LocalDateTime endDate;
```    

this two atribbutes have this format:"2007-12-03T10:15:30"  so with the addition of startHour and endHour can  both be erased(create a new attribute only with the date?)

Comment (fungame2270) [reply backend/uasmartsignage/src/main/java/deti/uas/uasmartsignage/Models/Schedule.java:55]: needs to be passed only to date 
the list of weekdays is ints
the start time and end hour is for daily use, the startDate and endDate is for use when the rule starts to be used

**Pull request merged**